include/grpc/grpc_security.h

```diff
@@ -528,9 +528,9 @@ GRPCAPI grpc_server_credentials* grpc_ssl_server_credentials_create(
    authenticate client certificates.*/
 GRPCAPI grpc_server_credentials* grpc_ssl_server_credentials_create_ex(
     const char* pem_root_certs, grpc_ssl_pem_key_cert_pair* pem_key_cert_pairs,
-    size_t num_key_cert_pair,
+    size_t num_key_cert_pairs,
     grpc_ssl_client_certificate_request_type client_certificate_request,
-    void* reserved);
+    bool send_client_ca_list, void* reserved);
```

Comment:
In this particular case, I don't think we want to add an argument to this existing method, for several reasons.

The back-story here is that when we first adding some of these options to SslCreds, there was some confusion about our policy with regard to backward-incompatible changes in the C-core API.  Our actual policy is that non-backward-compatible changes are fine, and the only requirement is that we need to publish a gRFC, as per [gRFC P3](https://github.com/grpc/proposal/blob/master/P3-grfcs-for-core-api-changes.md).  However, when these changes were added, for some reason we were thinking that we couldn't change the original `grpc_ssl_server_credentials_create()` API, so we added this `grpc_ssl_server_credentials_create_ex()` API with an alternative set of arguments.  And then later, we tried to generalize it more by adding `grpc_ssl_server_credentials_create_with_options()` (which is a good approach, and you'll notice that's the only interface we provide in the newer TlsCreds API).  But the result of all of this is that we now have 3 different methods for creating `grpc_ssl_server_credentials`, which is awful -- and if it wasn't for the fact that our long-term goal is to deprecate SslCreds in favor of TlsCreds, I would be pushing to clean this up.

The bottom line here is that we can make breaking changes, but it will require a gRFC, which has an inherent 10-day waiting period, which I don't think is what you want here.  (Technically, we should require a gRFC even for a non-breaking pure addition to the API, such as adding the `grpc_ssl_server_credentials_options_set_send_client_ca_list()` method.  But I'm willing to ignore that in this case, since (a) it's really unlikely that this addition will prove problematic or controversial, and (b) I don't care so much about long-term consequences, since we want to eventually deprecate SslCreds anyway.  But I am not comfortable making a breaking change to an existing API without a gRFC.)

Also, since we're going to want this PR back-ported to 1.56, we can't have any breaking changes here anyway, since that would require bumping the major version of C-core.  We generally only bump the major version of C-core in between minor versions, not in between two patch versions of the same minor version.

So, please don't add this parameter to this method.  Instead, just change any caller using `grpc_ssl_server_credentials_create_ex()` that needs to set this option to instead use `grpc_ssl_server_credentials_create_with_options()`.

Reply:
Thanks for the explanation, that makes sense. For the sake of getting this patch out quickly, I'm going to remove the changes to SslCreds from this PR (we can add this knob in a future PR) and turn the focus to TlsCreds only.

Reply:
Thanks @markdroth, the PR is ready for another pass. :)